tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/rules/GroovyScriptExecutor.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * (c) 2003-2017 MuleSoft, Inc. This software is protected under international copyright
+ * law. All use of this software is subject to MuleSoft's Master Subscription Agreement
+ * (or other master license agreement) separately entered into in writing between you and
+ * MuleSoft. If such an agreement is not in place, you may not use the software.
+ */
+package org.mule.test.infrastructure.process.rules;
+
+import org.mule.runtime.core.api.util.IOUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import groovy.lang.GroovyShell;
+
+/**
+ * Junit rule to run Groovy scripts.
+ */
+public class GroovyScriptExecutor extends ExternalResource {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MuleDeployment.class);
+
+  private final String scriptPath;
+  private Map<String, String> properties = new HashMap<>();
```

Comment:
Good point. Nevertheless, "properties" is the concept used by the `GroovyShell` class. Take a look at the method used for this matter. [`public void setProperty(String property, Object newValue) `](https://github.com/groovy/groovy-core/blob/master/src/main/groovy/lang/GroovyShell.java#L137). So, taking that into account, I mean, the fact that properties for the Groovy shell are name and value pair, I think we'd change this part. What do you think?